staging/cse/windows/nvidiagpudriverfunc.ps1

```diff
@@ -0,0 +1,154 @@
+function Start-InstallGPUDriver {
+    param(
+        [Parameter(Mandatory = $true)]
+        [bool]$EnableInstall,
+        # when the vm size does not have gpu, this value is an empty string.
+        [Parameter(Mandatory = $false)]
+        [string]$GpuDriverURL
+    )
+  
+    if (-not $EnableInstall) {
+        Write-Log "ConfigGPUDriverIfNeeded is false. GPU driver installation skipped as per configuration."
+        return
+    }
+  
+    Write-Log "ConfigGPUDriverIfNeeded is true. GPU driver installation started as per configuration."
+  
+    if ([string]::IsNullOrEmpty($GpuDriverURL)) {
+        Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_GPU_DRIVER_INSTALLATION_URL_NOT_SET -ErrorMessage "DriverURL is not properly specified."
+    }
+
+    $fileName = [IO.Path]::GetFileName($GpuDriverURL)
+    if (-not $fileName.EndsWith(".exe")) {
+        Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_GPU_DRIVER_INSTALLATION_URL_NOT_EXE -ErrorMessage "DriverURL does not point to a exe file"
+    }
+
+    $Target = "C:\AzureData\$fileName"
+    $LogFolder = "C:\AzureData"
+
+    Write-Log "Attempting to install Nvidia driver..."
+
+    Write-Log "Downloading from $GpuDriverURL to $Target"
+    DownloadFileOverHttp -Url $GpuDriverURL -DestinationPath $Target -ExitCode $global:WINDOWS_CSE_ERROR_GPU_DRIVER_INSTALLATION_DOWNLOAD_FAILURE
+    Write-Log "Installer download complete"
+    
+    $IsSignatureValid = VerifySignature $Target 
+    if ($IsSignatureValid -eq $false) {
+        Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_GPU_DRIVER_INVALID_SIGNATURE -ErrorMessage "Signature embedded in $($Target) is not valid."
+    }
+    else {
+        Write-Log "Signature embedded in $($Target) is valid."
```

Comment:
Duplicate logs since line 134 has logged it.

Reply:
I just realized $IsSignatureValid will also capture VerifySignature's Write-Output content. And its value will never be false. To fix the issue, I will move the Set-ExitCode inside the VerifySignature.